code/addons/docs/src/preset.ts

```diff
@@ -129,23 +129,36 @@ async function webpack(
   return result;
 }
 
-const storyIndexers: StorybookConfig['storyIndexers'] = (existingIndexers) => {
-  const mdxIndexer: StoryIndexer['indexer'] = async (fileName, opts) => {
-    let code = (await fs.readFile(fileName, 'utf-8')).toString();
-    const { compile } = global.FEATURES?.legacyMdx1
-      ? await import('@storybook/mdx1-csf')
-      : await import('@storybook/mdx2-csf');
-    code = await compile(code, {});
-    return loadCsf(code, { ...opts, fileName }).parse();
-  };
-  return [
-    {
-      test: /(stories|story)\.mdx$/,
-      indexer: mdxIndexer,
+const indexers: StorybookConfig['indexers'] = (existingIndexers) => [
+  {
+    test: /(stories|story)\.mdx$/,
+    index: async (fileName, opts) => {
+      let code = (await fs.readFile(fileName, 'utf-8')).toString();
+      const { compile } = global.FEATURES?.legacyMdx1
+        ? await import('@storybook/mdx1-csf')
+        : await import('@storybook/mdx2-csf');
+      code = await compile(code, {});
+      const csf = loadCsf(code, { ...opts, fileName }).parse();
+
+      // eslint-disable-next-line no-underscore-dangle
+      return Object.entries(csf._stories).map(([exportName, story]) => {
+        const docsOnly = story.parameters?.docsOnly;
+        const tags = (story.tags ?? csf.meta.tags ?? []).concat(docsOnly ? 'docsOnly' : []);
```

Comment:
Do these stories end up with the `stories-mdx` tag still?

Reply:
yes, [`@storybook/mdx2-csf` takes care of that](https://github.com/storybookjs/mdx2-csf/blob/2fa8864e919bec8f2d15f7a5eed715e64212f29c/src/sb-mdx-plugin.ts#L320C1-L320C1).

Although if tags are set on a story, it would overwrite it. It has always been like that, should I fix it?

Reply:
I don't know if you can set tags in `.stories.mdx` files, but maybe you should to be safe?